agent/taskresource/ssmsecret/ssmsecret.go

```diff
@@ -348,6 +348,15 @@ func (secret *SSMSecretResource) retrieveSSMSecretValues(region string, names []
 	}
 }
 
+// GetSecretValue returns the value retrieved from SSM
+func (secret *SSMSecretResource) GetSecretValue(key string) (string, bool) {
```

Comment:
There is another function in ssmsecret.go called getCachedSecretValue is doing the same thing as the function you added here. You may want to only keep one. Either remove that one or make that function public and remove this one. No matter what changes you made, do remember to update places calling getCachedSecretValue.

Reply:
wow how did i miss that.